ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/UtxoLedgerService.java

```diff
@@ -115,6 +117,46 @@ public interface UtxoLedgerService {
     @Suspendable
     <T extends ContractState> List<StateAndRef<T>> findUnconsumedStatesByExactType(@NotNull Class<T> type);
 
+    /**
+     * Finds unconsumed states of the specified {@link ContractState} type in the vault.
+     * <p>
+     * Example usage:
+     * <ul>
+     * <li>Kotlin:<pre>{@code
+     * val resultSet = utxoLedgerService.findUnconsumedStatesByExactType(MyState::class.java, 10, Instant.now())
+     *
+     * processResultsWithApplicationLogic(resultSet.results)
+     *
+     * while (resultSet.hasNext()) {
+     *     val results = resultSet.next()
+     *     processResultsWithApplicationLogic(results)
+     * }
+     * }</pre></li>
+     * <li>Java:<pre>{@code
+     * PagedQuery.ResultSet<StateAndRef<MyState>> resultSet = utxoLedgerService.query(MyState.class, 10, Instant.now())
+     *
+     * processResultsWithApplicationLogic(resultSet.getResults());
+     *
+     * while (resultSet.hasNext()) {
+     *     List<Integer> results = resultSet.next();
+     *     processResultsWithApplicationLogic(results);
+     * }
+     * }</pre></li>
+     *
+     * @param <T>   The underlying {@link ContractState} type.
+     * @param type  The {@link ContractState} type to find in the vault.
+     * @param limit The size of each page.
+     * @param createdTimestampLimit The timestamp limit the underlying query enforces.
+     * @return Returns a {@link ResultSet} of {@link StateAndRef} of unconsumed states of the specified type.
+     */
+    @NotNull
+    @Suspendable
+    <T extends ContractState> ResultSet<StateAndRef<T>> findUnconsumedStatesByExactType(
+            @NotNull Class<T> type,
+            @NotNull Integer limit,
+            @NotNull Instant createdTimestampLimit
```

Comment:
It keeps its functionality in line with the vault named query functionality. I think it makes sense to restrict it in the same way we do there, especially since the underlying mechanism is the same between them. We can do it internally if we wanted to, but either way it needs setting whether publicly or internally.

Reply:
Ok, then the question is whether we want to force users to have to provide it, or we provide another overload / a default argument that just sets it to now?

Reply:
I don't like the fact users have to provide a timestamp on this API - there should at least be an override that defaults to `now()`.